File: frontend/src/types/transaction.ts
export type TransactionStatus = 'Pending' | 'Completed' | 'Failed';

export interface Transaction {
  transactionId: string;
  amount: number;
  currency: string;
  status: TransactionStatus;
  timestamp: string;
}

export interface TransactionFormData {
  amount: number;
  currency: string;
  status: TransactionStatus;
}

File: backend/FinancialMonitor.Api/Models/Transaction.cs
namespace FinancialMonitor.Api.Models;

public record Transaction
{
    public string TransactionId { get; init; } = string.Empty;
    public decimal Amount { get; init; }
    public string Currency { get; init; } = string.Empty;
    public TransactionStatus Status { get; init; }
    public DateTimeOffset Timestamp { get; init; }
}
